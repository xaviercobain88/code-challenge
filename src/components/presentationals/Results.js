import React from 'react';
import PropTypes from 'prop-types';
import { ScrollView } from 'react-native';
import { Feather } from '@expo/vector-icons';
import styled from 'styled-components';
import { H1 } from './StyledText';
import Card from './Card';
import Button from './Button';
import ButtonWrapper from './ButtonWrapper';

const Container = styled.View`
  flex: 1;
  align-items: center;
`;

const ResultWrapper = styled.View`
  flex: 1;
  flex-direction: row;
  justify-content: flex-start;
`;

const Text = styled.Text`
  flex: 1;
  font-size: 15;
  margin-left: 10;
  margin-bottom: 10;
`;

const Result = ({ isCorrect, question }) => (
  <Card>
    <ResultWrapper>
      <Feather
        name={isCorrect ? 'check' : 'x'}
        size={40}
        color={isCorrect ? 'green' : 'red'}
      />
      <Text>
        {question}
      </Text>
    </ResultWrapper>
  </Card>
);

Result.propTypes = {
  isCorrect: PropTypes.bool.isRequired,
  question: PropTypes.string.isRequired,
};

const Results = (props: {
  score: string,
  questions: Array<any>,
  onPlayAgain: Function
}) => (
  <Container>
    <H1>
You scored
    </H1>
    <H1>
      {props.score}
    </H1>

    <ScrollView>
      {props.questions.map(question => (
        <Result key={question.index} {...question} />
      ))}
    </ScrollView>
    <ButtonWrapper>
      <Button
        inverted
        onPress={props.onPlayAgain}
        title="PLAY AGAIN!"
        color="white"
      />
    </ButtonWrapper>
  </Container>
);

export default Results;
