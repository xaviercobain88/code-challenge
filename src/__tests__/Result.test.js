import React from 'react';
import renderer from 'react-test-renderer';
import Results from '../components/presentationals/Results';
import 'jest-styled-components';

const props = {
  score: '',
  questions: [],
  onPlayAgain: () => ({}),
};

test('renders correctly', () => {
  const tree = renderer.create(<Results {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
