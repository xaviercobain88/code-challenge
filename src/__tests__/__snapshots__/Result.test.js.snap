// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 30,
          "fontWeight": "600",
          "paddingBottom": 10,
          "paddingTop": 20,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    You scored
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 30,
          "fontWeight": "600",
          "paddingBottom": 10,
          "paddingTop": 20,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    
  </Text>
  <RCTScrollView>
    <View />
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 72,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      color="white"
      inverted={true}
      onPress={[Function]}
      style={
        Array [
          Object {
            "borderColor": "white",
            "borderRadius": 30,
            "borderWidth": 2,
            "marginHorizontal": 10,
            "marginVertical": 10,
            "paddingVertical": 5,
            "width": 120,
          },
          undefined,
        ]
      }
      title="PLAY AGAIN!"
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            PLAY AGAIN!
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
